### PR TITLE
Add backup spack mirror on scisoft.fnal.gov

### PR DESCRIPTION
**Pull request by gartung**

### CODE REVIEW

**Comment by gartung** [comment]
Adding the backup mirror didn't help an offsite install because the bootstrap patchelf could not be downloaded.

File: bin/make_spack [hunk 1]
```diff
@@ -49,7 +49,9 @@ bootstrap_patchelf() {
     export pdir
     mkdir $pdir
     PATH="$pdir:$PATH"
-    wget https://spack-cache-1.fnal.gov/binaries/bootstrap/patchelf.linux-scientific7-x86_64
+    curl -OL https://spack-cache-1.fnal.gov/binaries/bootstrap/patchelf.linux-scientific7-x86_64
```
Comment (greenc-FNAL) [inline bin/make_spack]: I would add `-J --fail` to the curl options and deal with an error condition here.

File: bin/make_spack [hunk 2]
```diff
@@ -121,6 +123,8 @@ create_spack() {
             export plat="target=`uname -i` os=`spack arch --operating-system`"
             spack mirror add --scope site fnal $binary_cache
             spack buildcache update-index  -d $binary_cache
+            spack mirror add --scope site scisoft $binary_cache_bak
+            spack buildcache update-index  -d $binary_cache_bak
```
Comment (greenc-FNAL) [inline bin/make_spack:125]: Verify these commands work with older and newer Spack, please?

File: bin/make_subspack [hunk 3]
```diff
@@ -155,6 +156,9 @@ echo "adding package repos..."
 echo "Adding buildcache $binary_cache"
 $SPACK_ROOT/bin/spack mirror add --scope site fnal $binary_cache
 $SPACK_ROOT/bin/spack buildcache update-index  -d $binary_cache
+echo "Adding backup buildcache $binary_cache_bak"
+$SPACK_ROOT/bin/spack mirror add --scope site scisoft $binary_cache_bak
+$SPACK_ROOT/bin/spack buildcache update-index  -d $binary_cache_bak
```
Comment (greenc-FNAL) [inline bin/make_subspack:161]: Verify these commands work with older and newer Spack, please?

**Review by greenc-FNAL — CHANGES_REQUESTED** [review]

File: bin/make_spack (hunk 1 above)
Comment (gartung) [reply bin/make_spack]: The error code when I run this from CERN is 0 and the file is text.
```
 curl -OJL --fail https://spack-cache-1.fnal.gov/binaries/bootstrap/patchelf.linux-scientific7-x86_64
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0   417    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0  1341    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
100  9174  100  9174    0     0   4904      0  0:00:01  0:00:01 --:--:--  4904
-bash-4.2$ echo $?
0
```

Comment (gartung) [reply bin/make_spack]: If I omit `-L` I still get a text file and error code 0
```
  curl -OJ --fail https://spack-cache-1.fnal.gov/binaries/bootstrap/patchelf.linux-scientific7-x86_64
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   417  100   417    0     0    582      0 --:--:-- --:--:-- --:--:--   582
-bash-4.2$ echo $?
0
-bash-4.2$ cat patchelf.linux-scientific7-x86_64
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>303 See Other</title>
</head><body>
<h1>See Other</h1>
<p>The answer to your request is located <a href="https://spack-cache-1.fnal.gov/mellon/login?ReturnTo=https%3A%2F%2Fspack%2Dcache%2D1.fnal.gov%2Fbinaries%2Fbootstrap%2Fpatchelf.linux%2Dscientific7%2Dx86_64&amp;IdP=https%3A%2F%2Fidp.fnal.gov%2Fidp%2Fshibboleth">here</a>.</p>
</body></html>
```

Comment (greenc-FNAL) [reply bin/make_spack]: > The error code when I run this from CERN is 0 and the file is text.
> 
> ```
>  curl -OJL --fail https://spack-cache-1.fnal.gov/binaries/bootstrap/patchelf.linux-scientific7-x86_64
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
>   0   417    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
>   0  1341    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
> 100  9174  100  9174    0     0   4904      0  0:00:01  0:00:01 --:--:--  4904
> -bash-4.2$ echo $?
> 0
> ```

What is the content of the text file here?

Comment (gartung) [reply bin/make_spack]: Much longer with instructions for accessing.

Comment (greenc-FNAL) [reply bin/make_spack]: Can you pull it back to scisoftbuild01 so I can see, please?

Comment (gartung) [reply bin/make_spack]: Done `~gartung/patchelf.linux-scientific7-x86_64`

Comment (greenc-FNAL) [reply bin/make_spack]: OK. The following curl command will fail with code 47 if the requested file is redirected (_e.g._ to the authorization page):
```
curl -OJL --max-redirs 0 --silent --fail https://spack-cache-1.fnal.gov/binaries/bootstrap/patchelf.linux-scientific7-x86_64
```

File: bin/make_spack [hunk 4]
```diff
@@ -49,7 +49,7 @@ bootstrap_patchelf() {
     export pdir
     mkdir $pdir
     PATH="$pdir:$PATH"
-    wget https://spack-cache-1.fnal.gov/binaries/bootstrap/patchelf.linux-scientific7-x86_64
+    (curl -OJL --max-redirs 0 --silent --fail https://spack-cache-1.fnal.gov/binaries/bootstrap/patchelf.linux-scientific7-x86_64 || curl -OJL --max-redirs 0 --silent --fail  https://scisoft.fnal.gov/scisoft/spack-mirror/patchelf.linux-scientific7-x86_64) || (echo "patchelf bootstrap download failed"; exit 111)
```
Comment (greenc-FNAL) [inline bin/make_spack]: Any reason you're using hardwired URLs here instead of `$binary_cache` and `$binary_cache_bak`?

**Review by marcmengel — APPROVED** [review]

**Review by greenc-FNAL — CHANGES_REQUESTED** [review]

File: bin/make_spack (hunk 4 above)
Comment (greenc-FNAL) [reply bin/make_spack]: Sorry, wrote this yesterday but apparently forgot to complete the review.

Comment (gartung) [reply bin/make_spack]: Done

**Review by greenc-FNAL — APPROVED** [review]

**Pull request merged**